src/javascript/Validation/validation.js

```diff
@@ -55,6 +55,21 @@ const dateFieldValidation = (values, field) => {
     return respectEachConstraint ? undefined : error;
 };
 
+const patternFieldValidation = (values, field) => {
+    const error = 'invalidPattern';
+
+    const constraints = field.valueConstraints && field.valueConstraints.map(constraint => constraint.value.string);
+
+    if (constraints && constraints.length > 0 && field.requiredType === 'STRING') {
+        const fieldValues = field.multiple ? values[field.name] : [values[field.name]];
+
+        // If one pattern is invalid, error!
+        if (fieldValues.some(emailValue => !RegExp(constraints[0]).test(String(emailValue).toLowerCase()))) {
+            return error;
```

Comment:
Why `constraints[0]`? JCR Spec says that the value must match a least one of the constraints. 
Please rename emailValue to something less generic.

Reply:
Ah, you'd like to say, we must validate all the constraints? and should all be regex?